devices/debian_wifi.py

```diff
@@ -7,29 +7,38 @@ class DebianWifi(debian.DebianBox):
 
     model = ('debianwifi')
 
-    iface_wlan = "wlan1"
-
-    def scan(self):
+    def wifi_scan(self, iface_wlan="wlan1", login_pwd=None):
         from tests.lib.installers import install_iw
         install_iw(self)
 
-        self.sendline('iw %s scan | grep SSID:' % self.iface_wlan)
+        self.sendline('iw %s scan | grep SSID:' % iface_wlan)
         self.expect(self.prompt)
-
+        match = re.search("Operation not permitted", self.before)
+        if match:
+            self.sendline('sudo iw %s scan | grep SSID:' % iface_wlan)
+            idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
+            if idx == 0:
+                self.sendline(login_pwd)
+                self.expect(self.prompt)
         return self.before
 
-    def wpa_connect(self, ssid_name, password):
+    def wpa_connect(self, ssid_name, password, iface_wlan="wlan1", login_pwd=None):
```

Comment:
What is the iface for? Don't we have self.iface?

Is this for two wireless interfaces?

Reply:
Yes we have 2 interfaces. Added them in Debianwifi class